src/systems/model_photo_shoot/ModelPhotoShoot.cc

```diff
@@ -0,0 +1,283 @@
+
+  /// \brief Boolean to control we only take the pictures once.
+  public: bool takePicture{true};
+
+  /// \brief Boolean to control if joints should adopt random poses.
+  public: bool randomPoses{false};
+
+  /// \brief File to save translation and scaling info.
+  public: std::ofstream savingFile;
+};
+
+//////////////////////////////////////////////////
+ModelPhotoShoot::ModelPhotoShoot()
+    : System(), dataPtr(std::make_unique<ModelPhotoShootPrivate>())
+{
+}
+
+//////////////////////////////////////////////////
+void ModelPhotoShoot::Configure(const ignition::gazebo::Entity &_entity,
+                                const std::shared_ptr<const sdf::Element> &_sdf,
+                                ignition::gazebo::EntityComponentManager &_ecm,
+                                ignition::gazebo::EventManager &_eventMgr)
+{
+  std::string save_data_location =
+      _sdf->Get<std::string>("translation_data_file");
+  if (save_data_location.empty())
+  {
+    igndbg << "No data location specified, skipping translaiton data"
+              "saving.\n";
+  }
+  else
+  {
+    igndbg << "Saving translation data to: "
+        << save_data_location << std::endl;
+    this->dataPtr->savingFile.open(save_data_location);
+  }
+
+  if (_sdf->HasElement("random_joints_pose"))
+  {
+    this->dataPtr->randomPoses = _sdf->Get<bool>("random_joints_pose");
+  }
+
+  this->dataPtr->connection =
+      _eventMgr.Connect<ignition::gazebo::events::PostRender>(std::bind(
+          &ModelPhotoShootPrivate::PerformPostRenderingOperations,
+          this->dataPtr.get()));
+
+  this->dataPtr->model = std::make_shared<ignition::gazebo::Model>(_entity);
+  this->dataPtr->modelName = this->dataPtr->model->Name(_ecm);
+  // Get the pose of the model
+  this->dataPtr->modelPose3D =
+      ignition::gazebo::worldPose(this->dataPtr->model->Entity(), _ecm);
+}
+
+//////////////////////////////////////////////////
+void ModelPhotoShoot::PreUpdate(
+    const ignition::gazebo::UpdateInfo &,
+    ignition::gazebo::EntityComponentManager &_ecm)
+{
+  if (this->dataPtr->randomPoses)
+  {
+    std::vector<gazebo::Entity> joints = this->dataPtr->model->Joints(_ecm);
+    unsigned seed =
+        std::chrono::system_clock::now().time_since_epoch().count();
+    std::default_random_engine generator(seed);
+    for (auto joint = joints.begin(); joint != joints.end(); ++joint)
+    {
+      auto jointNameComp = _ecm.Component<components::Name>(*joint);
+      if (jointNameComp && jointNameComp->Data() != "World")
+      {
+        // Using the JointAxis component to extract the joint pose limits
+        auto jointAxisComp = _ecm.Component<components::JointAxis>(*joint);
+        if (jointAxisComp)
+        {
+          float mean =
+              (jointAxisComp->Data().Lower() +
+              jointAxisComp->Data().Upper()) / 2;
+          float stdv = (jointAxisComp->Data().Upper() - mean) / 3;
+          std::normal_distribution<float> distribution(mean, stdv);
```

Comment:
It's not clear to me how the `stdv` is being calculated. Besides, a normal distribution generate random values outside the joint limit range. How about using a `std::uniform_real_distribution` instead?

Reply:
I wanted to avoid very extreme positions as I thought they might not be the most usual ones for robots and it would be more realistic to just test using a `normal_distribution` for the positions. The `stdv` is set so most of these random values lie in the middle area of the distribution. But I have nothing against using a `std::uniform_real_distribution` instead so I will go ahead and change it.